Make OneNord follow background option as it's set

Slotos opened the pull request:
Before this change, the option was followed once
on initial load due to `require` behaviour.

Additionally, theme option is now respected
on repeated loads. E.g.:

```
:lua require('onenord').setup({theme = 'dark'})
:lua require('onenord').setup({theme = 'light'})
```

Yields a light variant, as you'd expect.

In order to unset forced theme, set it to `false`.
This is due to the fact that `{a = nil}` and `{}`
are equivalent. As such, providing `theme = nil`
is equivalent to not providing it at all.

Slotos commented:
Forced push fixes StyLua [style violation](https://github.com/rmehri01/onenord.nvim/runs/4730453621?check_suite_focus=true#step:3:13).

rmehri01 reviewed (approved):
Hi, thanks for the PR, I think it looks good! Do you mind adding `BREAKING CHANGE:` in front of your commit message just so people know when they upgrade?

Slotos commented:
Added `breaking change`, as it's not really a fix, but a change that might break some workflows.

I also removed `TODO:` part, replacing it with explanation about why `theme = nil` doesn't reset theme when calling `setup`.

rmehri01 commented:
I see, thanks again!
